api.py

```diff
@@ -176,36 +175,83 @@ def bulk():
         logging.debug("No data recieved dropping")
         return ""
 
-    start_time = datetime.fromtimestamp(int(request.args.get('time')))
-    for row in data:
-
+    for row in data: 
         inserts = dict()
         inserts['timestamp'] = datetime.fromtimestamp(int(row[0]))
         inserts['site_id'] = g.account['site_id']
         inserts['site_name'] = g.site['site_name']
-        logging.debug("got post in get "+str(inserts))
         node_id = int(row[1]) # Adding distinction between nodes
-        table = None
-
-        if app.config['BULK_INDEX_MAPPING'].has_key(node_id):
-
+        bulk_index_mapping = generate_bulk_index_conf(g.site)
 
-            for index in app.config["BULK_INDEX_MAPPING"][node_id]:
-                if isinstance(index,int) and len(row) >= index+1: # Make sure we have an entry for that index. just in case
-                    inserts[app.config['BULK_INDEX_MAPPING'][node_id][index]] = row[index]
+        if bulk_index_mapping.has_key(node_id):
+            length_of_indices = len(row)
+            sensor_data = bulk_index_mapping[node_id]
 
-                # Find out which table the data needs to goto
-                table = app.config['BULK_INDEX_MAPPING'][node_id]['table']
-            logging.debug("inserting %s into table %s"%(inserts,table))
+            # Generating an insert dictionary representing the mapping of the data to the sensor mapping(bulk_index_mappng)
+            for index, item_name in sensor_data.items():
+                if length_of_indices > index: # Make sure we have an entry for that index. just in case
+                    inserts[item_name] = row[index]
 
-            # We need to send the data to the correct table according to the type of data it is
-            insert_data(inserts, table=table, mysql=app.config['MYSQL_INSERT'])
+            logging.debug("The data to be written to the database is: %s" % inserts)
+            insert_data(inserts)
         else:
-            logging.warning("No table mapping found for in BULK_INDEX_MAPPING nodeid=%s skipping"%node_id)
+            logging.warning("The node_id=%s sent has no mapping for this site. SKIPPING" % node_id)
 
     return "OK"
 
 
+
+def get_associated_sensors(site):
+    """
+    Queryies and returns associated sensors for a given
+    site
+    """
+    associated_sensors = []
+    sensor_table_names = ['seshdash_sensor_bmv', 'seshdash_sensor_emontx', 'seshdash_sensor_emonth']
```

Comment:
use seshdash_sensor_mapping table from the dashboard to get the table names.

to get the table type query the `select sensor_type from sensor_mapping where site_id = g.site_id`

https://github.com/GreatLakesEnergy/sesh-dash-beta/blob/master/seshdash/models.py

then based on the node_id you'll lookup the table name